/*
Copyright 2020 The Magma Authors.

This source code is licensed under the BSD-style license found in the
LICENSE file in the root directory of this source tree.

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

syntax = "proto3";

package magma.orc8r;
option go_package = "magma/orc8r/lib/go/protos";

message Void {
}

// -------------------------------------------------------------------------------
// Bytes is a special message type used to marshal & unmarshal unknown types as is
// -------------------------------------------------------------------------------
message Bytes {
  bytes val = 1;
}

// --------------------------------------------------------------------------
// NetworkID uniquely identifies the network
// --------------------------------------------------------------------------
message NetworkID {
  string id = 1;
}

// --------------------------------------------------------------------------
// IDList is a generic definition of an array of IDs (network, gateway, etc.)
// --------------------------------------------------------------------------
message IDList {
  repeated string ids = 1;
}

// --------------------------------------------------------------------------
// Logging levels
// --------------------------------------------------------------------------
enum LogLevel {
  DEBUG = 0;
  INFO = 1;
  WARNING = 2;
  ERROR = 3;
  FATAL = 4;
}
